src/Symfony/Component/Lock/Store/RedisStore.php

```diff
@@ -38,7 +38,7 @@ class RedisStore implements StoreInterface
      */
     public function __construct($redisClient, float $initialTtl = 300.0)
     {
-        if (!$redisClient instanceof \Redis && !$redisClient instanceof \RedisArray && !$redisClient instanceof \RedisCluster && !$redisClient instanceof \Predis\ClientInterface && !$redisClient instanceof RedisProxy) {
+        if (!$redisClient instanceof \Redis && !$redisClient instanceof \RedisArray && !$redisClient instanceof \RedisCluster && !$redisClient instanceof \Predis\ClientInterface && !$redisClient instanceof RedisProxy && !$redisClient instanceof RedisClusterProxy) {
```

Comment:
That were missing on purpose. see my previous comment. 
From a concurrency point of view, using a cluster is less reliable than a single node.

Reply:
I understand and using CombinedStore would remain as the recommended way in documentation. But from a point of consistency, one would expect RedisStore and StoreFactory to work with same DSN syntax as AbstractAdapter since that is being used in the background. For me this seems like unexpected behaviour.

Reply:
Well, if the RedisStore does not support clusters on purpose, I don't see why we should add it just for consistency. Consistency is good when you want that 2 implementations supporting clusters do it with the same DSN syntax. But it is not a reason to support non-working features IMO.

Reply:
I will be happy to review and approve a PR that add `WAIT` in RedisStore. 
That would fix the reliability issue, and allows us to use RedisCluster connection.

Reply:
As I understood, lock store will work even when using a cluster but it just is not as reliable in case of failrue? This is already documented and user has been warned but they would have the option.